test_simulation.cpp

```diff
@@ -171,6 +172,164 @@ int test_with_reduced_stochasticity()
+        cout << "Deterministic Kernel Cauchy: mortality tracker (actual, expected):\n" << mortality_tracker << "  !=\n" << expected_mortality_tracker << "\n";
+        return 1;
+    }
+    cout << "Deterministic Kernel Cauchy: no errors\n";
+    
+    infected = {{5, 0, 0}, {0, 5, 0}, {0, 0, 2}};
+    mortality_tracker = {{0, 0, 0}, {0, 0, 0}, {0, 0, 0}};
+    susceptible = {{10, 20, 9}, {14, 15, 0}, {3, 0, 2}};
+    total_hosts = susceptible;
+    movements = {{0, 0, 1, 1, 2}, {0, 1, 0, 0, 3}, {0, 1, 1, 0, 2}};
+    movement_schedule = {1, 1};
+	std::vector<std::tuple<int, int>> outside_dispersers2;
+    expected_mortality_tracker = {{10, 0, 0}, {0, 10, 0}, {0, 0, 2}};
+    expected_infected = {{15, 0, 0}, {0, 15, 0}, {0, 0, 4}};
+    dispersers(infected.rows(), infected.cols());
+    
+    // Exponential
+    Simulation<Raster<int>, Raster<double>> s2(
+                42,
+                infected.rows(),
+                infected.cols(),
+                model_type_from_string("SI"),
+                0,
+                generate_stochasticity,
+                establishment_stochasticity
+                );
+    s2.generate(dispersers, infected, weather, weather_coefficient, reproductive_rate);
+    expected_dispersers = reproductive_rate * infected;
+    if (dispersers != expected_dispersers) {
+        cout << "Deterministic Kernel Exponential: dispersers (actual, expected):\n" << dispersers << "  !=\n" << expected_dispersers << "\n";
+        return 1;
+    }
+    RadialDispersalKernel deterministicKernel2(30, 30, DispersalKernelType::Exponential, 1.0, Direction::None, 0.0, true, dispersers, 0.99, 0.0);
+   
+    s2.disperse(dispersers, susceptible, infected,
+                        mortality_tracker, total_hosts,
+                        outside_dispersers2, weather, weather_coefficient,
+                        deterministicKernel2, establishment_probability);
+    if (outside_dispersers2.size() != 0) {
+        cout << "Deterministic Kernel Exponential: There are outside_dispersers (" << outside_dispersers2.size() << ") but there should be 0\n";
+        return 1;
+    }
+    if (infected != expected_infected) {
+        cout << "Deterministic Kernel Exponential: infected (actual, expected):\n" << infected << "  !=\n" << expected_infected << "\n";
+        return 1;
+    }
+    if (mortality_tracker != expected_mortality_tracker) {
+        cout << "Deterministic Kernel Exponential: mortality tracker (actual, expected):\n" << mortality_tracker << "  !=\n" << expected_mortality_tracker << "\n";
+        return 1;
+    }
+    cout << "Deterministic Kernel Exponential: no errors\n";
+    
+    // testing cauchy pdf & cdf
+    double scale = 0.001;  // rounding to thousands place
+    CauchyDistribution cauchy(1.0, 0.0);
+    double probability = (int) (cauchy.pdf(5) / scale) * scale;
+    if ( probability != 0.012) {
+    	cout << "Cauchy Distribution: probability was " << probability << " but should be 0.012\n";
```

Comment:
`0.012` will be better as a variable something like `probability_ref`(erence). Then you can also do `<< probability_ref`. More characters, but only one place where the number is. (Same for the others.)